src/schema/v2/phoneNumber.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  GraphQLBoolean,
+  GraphQLEnumType,
+  GraphQLFieldConfig,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { ResolverContext } from "types/graphql"
+import {
+  PhoneNumberUtil,
+  PhoneNumber as GooglePhoneNumber,
+  PhoneNumberFormat,
+} from "google-libphonenumber"
+
+interface PhoneNumberTypeSource {
+  phoneNumber: string
+  parsedPhone?: GooglePhoneNumber
+  phoneUtil: PhoneNumberUtil
+}
+
+export const PhoneNumberFormats = {
+  type: new GraphQLEnumType({
+    name: "PhoneNumberFormats",
+    values: {
+      INTERNATIONAL: {
+        value: PhoneNumberFormat.INTERNATIONAL,
+      },
+      E164: {
+        value: PhoneNumberFormat.E164,
+      },
+      NATIONAL: {
+        value: PhoneNumberFormat.NATIONAL,
+      },
+      RFC3966: {
+        value: PhoneNumberFormat.RFC3966,
+      },
+    },
+  }),
+}
+
+const PhoneNumberType: GraphQLObjectType<
+  PhoneNumberTypeSource,
+  ResolverContext
+> = new GraphQLObjectType<PhoneNumberTypeSource, ResolverContext>({
+  name: "PhoneNumberType",
+  fields: {
+    isValid: {
+      type: GraphQLBoolean,
+      resolve: ({ parsedPhone, phoneUtil }) =>
+        parsedPhone ? phoneUtil.isValidNumber(parsedPhone) : false,
+    },
+    errorMessage: {
+      type: GraphQLString,
+      resolve: ({ parsedPhone, phoneUtil }) => {
+        if (parsedPhone) {
+          switch (phoneUtil.isPossibleNumberWithReason(parsedPhone)) {
+            case PhoneNumberUtil.ValidationResult.TOO_SHORT:
+              return "Too short"
+
+            case PhoneNumberUtil.ValidationResult.TOO_LONG:
+              return "Too long"
+
+            case PhoneNumberUtil.ValidationResult.INVALID_COUNTRY_CODE:
+              return "Invalid country code"
+          }
+
+          if (!phoneUtil.isValidNumber(parsedPhone)) {
+            return "Invalid number"
```

Comment:
**suggestion:** Instead of returning a string, I think that it may be more useful for clients if Metaphysics returns an enum value from `{ TOO_SHORT, TOO_LONG, INVALID_COUNTRY_CODE, INVALID_NUMBER }`. What do you think?